CONTRIBUTING.md

```diff
@@ -1,5 +1,197 @@
-# Contributing
+# Contributing to JupyterLab
 
-We follow the [IPython Contributing Guide](https://github.com/ipython/ipython/blob/master/CONTRIBUTING.md).
+If you're reading this section, you're probably interested in contributing to
+JupyterLab.  Welcome and thanks for your interest in contributing!
+
+Please take a look at the Contributor documentation, familiarize yourself with
+using the Jupyter Notebook, and introduce yourself on the mailing list and share
+what area of the project you are interested in working on.
+
+## General Guidelines
+
+For general documentation about contributing to Jupyter projects, see the
+[Project Jupyter Contributor Documentation](http://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
 
 All source code is written in [TypeScript](http://www.typescriptlang.org/Handbook). See the [Style Guide](https://github.com/jupyter/jupyterlab/wiki/TypeScript-Style-Guide).
+
+
+## Setting Up a Development Environment
+
+### Installing Node.js and npm
+
+Building the Notebook from its GitHub source code requires some tools to
+create and minify JavaScript components and the CSS.
+Namely, that's Node.js and Node's package manager, ``npm``.
+
+If you use ``conda``, you can get them with:
+
+```bash
+conda install -c javascript nodejs
+```
+
+If you use [Homebrew](http://brew.sh/) on Mac OS X:
+
+```bash
+brew install node
+```
+
+For Debian/Ubuntu systems, you should use the ``nodejs-legacy`` package instead
+of the ``node`` package:
+
+```bash
+sudo apt-get update
+sudo apt-get install nodejs-legacy npm
+```
+
+You can also use the installer from the [Node.js](https://nodejs.org) website.
+
+
+## Installing JupyterLab
+
+JupyterLab requires Jupyter Notebook version 4.2 or later.
+
+If you use ``conda``, you can install it as:
+
+```bash
+conda install notebook
+```
+
+If you use ``pip``` you can install it as:
+
+```bash
+pip install notebook
+```
+
+Once you have installed the dependencies mentioned above, use the following
+steps::
+
+    git clone https://github.com/<your-github-username>/jupyterlab.git
```

Comment:
You need to fork it first if you do that :-) Should we just tell people to clone form jupyter/jupyterlab ?